nmap.go

```diff
@@ -117,52 +118,107 @@ func (s *Scanner) Run() (*Run, error) {
 }
 
 // RunAsync runs nmap asynchronously and returns error.
-func (s *Scanner) RunAsync() error {
+func (s *Scanner) RunAsync() (<-chan []byte, <-chan []byte, error) {
+	stdoutChannel := make(chan []byte)
+	stderrChannel := make(chan []byte)
+
 	// Enable XML output.
 	s.args = append(s.args, "-oX")
 
 	// Get XML output in stdout instead of writing it in a file.
 	s.args = append(s.args, "-")
 	s.cmd = exec.Command(s.binaryPath, s.args...)
 
+	// Get CMD Stderr Pipe
 	stderr, err := s.cmd.StderrPipe()
 	if err != nil {
-		return fmt.Errorf("unable to get error output from asynchronous nmap run: %v", err)
+		return nil, nil, fmt.Errorf("unable to get error output from asynchronous nmap run: %v", err)
 	}
 
+	// Get CMD Stdout Pipe
 	stdout, err := s.cmd.StdoutPipe()
 	if err != nil {
-		return fmt.Errorf("unable to get standard output from asynchronous nmap run: %v", err)
+		return nil, nil, fmt.Errorf("unable to get standard output from asynchronous nmap run: %v", err)
 	}
 
-	s.stdout = *bufio.NewScanner(stdout)
-	s.stderr = *bufio.NewScanner(stderr)
-
 	if err := s.cmd.Start(); err != nil {
-		return fmt.Errorf("unable to execute asynchronous nmap run: %v", err)
+		return nil, nil, fmt.Errorf("unable to execute asynchronous nmap run: %v", err)
 	}
 
+	// Stream stdout to the stdoutChannel
+	go func() {
```

Comment:
Since both functions to stream the output are very similar, with the only difference being the channel in which to write, and the file descriptor from which to read; what do you think about making a function called `streamFromFileDescriptor(in io.Writer, out <-chan []byte) error`, which would be called both times with the proper arguments for `stderr` and `stdout`?

Reply:
This would also make this section of the code much clearer! :)